### DISCUSSION TITLE
Do tempo-cli support tls or skip tls ?

### DISCUSSION THREAD

**Discussion by shalk** [grafana/tempo#4134] [General]
We have build a tempo cluster with tls, It works fine in most case. But I get some error on dashboard

```
failed to get trace with id: 134ce96213bf25ee5115c1ef93cce7 Status: 500 Internal Server Error Body: error querying ingesters in Querier.FindTraceByID: failed to execute f() for 10.247.0.10:9095: rpc error: code = Unknown desc = completeBlock.FindTraceByID failed: error retrieving bloom bloom-0 (abcd, 1c5280a4-485b-4dda-b507-cb5b0cbedaa1): does not exist
```

I want to use trace-query to do some debug
```
./tempo-cli-arm64 query api trace-id https://x.x.x.x:12200 134ce96213bf25ee5115c1ef93cce7
```
I get the error info
```
tempo-cli-arm64: error: main.queryTraceIDCmd.Run(): error querying Tempo Get "https://xxxxx:12200/api/traces/134ce96213bf25ee5115c1ef93cce7": tls: failed to verify certificate: x509: cannot validate certificate for xxxx because it doesn't contain any IP SANs
```
Do tempo cli support tls or can skip tls?

**Comment by joe-elliott** [comment]
I'm fairly certain that call lands here:

https://github.com/grafana/tempo/blob/main/pkg/httpclient/client.go#L131

This is the default go http client and so will likely behave by all https rules. You could make a temp change to this code by doing something like:

https://stackoverflow.com/questions/12122159/how-to-do-a-https-request-with-bad-certificate

We'd also accept a PR to add a switch to ignore bad certs if you want to work on it.
